tests/test_build.py

```diff
@@ -254,3 +255,96 @@ def test_included_toc(sphinx_build_factory):
     sphinx_build = sphinx_build_factory("test_included_toc").build()
     included_page_html = sphinx_build.html_tree("included-page.html")
     assert included_page_html is not None
+
+
+good_edits = [
+    [
+        {
+            "github_user": "foo",
+            "github_repo": "bar",
+            "github_version": "HEAD",
+            "doc_path": "docs",
+        },
+        "https://github.com/foo/bar/edit/HEAD/docs/index.rst",
+    ],
+    [
+        {
+            "gitlab_user": "foo",
+            "gitlab_repo": "bar",
+            "gitlab_version": "HEAD",
+            "doc_path": "docs",
+        },
+        "https://gitlab.org/foo/bar/edit/HEAD/docs/index.rst",
+    ],
+    [
+        {
+            "bitbucket_user": "foo",
+            "bitbucket_repo": "bar",
+            "bitbucket_version": "HEAD",
+            "doc_path": "docs",
+        },
+        "https://bitbucket.org/foo/bar/src/HEAD/docs/index.rst?mode=edit",
```

Comment:
need to verify this URL...

Reply:
Verified. It sometimes has a duplicate `&at=HEAD` at the end, but doesn't seem to really do anything.